### DISCUSSION TITLE
Loss of data points when using route_path()

### DISCUSSION THREAD

**Discussion by vheim** [ianjonsen/foieGras#67] [Q&A] [answered]
Hello,

I am using animotum to fit SSMs to Argos satellite data. This has been working nicely so far, but the newest data set comes from a location with lots of small islands and cays. As a consequence many location estimates overlap with land mass so that I decided to use route_path() to re-route locations off the land barriers. However, I noticed that this causes the deletion of observations compared to the non-rerouted SSMs.

The fit_ssm() runs smoothly without any issues/warnings with the following code:

```
mod.crw_pf <- fit_ssm(
  as.data.frame(dpf),
  spdf = F,
  model = "crw",
  time.step = NA
)
```

[Output_SSM.csv](https://github.com/user-attachments/files/16709247/Output_SSM.csv)

I use the following code for the re-routing steps:
```
mod.crw_pf_rr <- route_path(mod.crw_pf,
                            what = "fitted",
                            map_scale = 10, 
                            dist = 10000
                            append = T
                            )
```
This is the output:
[Output_SSM_rerouted.csv](https://github.com/user-attachments/files/16709271/Output_SSM_rerouted.csv)

The same is true if I use fit_ssm() to predict most probable locations at a time step of 12 hrs and re-routed the outputted tracks, albeit with the exception that even more observations are removed.

I am a bit stuck with finding a plausible explanation for this?

Thank you in advance,
Vital

**Comment by ianjonsen** [comment, marked as answer]
Hi, 

If the network of land barriers is complex it is possible that the underlying `pathroutr` functions cannot find a suitable solution. It is difficult for me to suggest a clear solution, if one can indeed be found, without having access to your input data (`dpf`) and reproducing the issue myself. Failing that, a couple of things to try are: 1) specify a short time.step, eg. 2 hours - but this will depend on the temporal resolution of your data; 2) add the following `pathroutr::prt_visgraph()` arguments to your `route_path()` call:

```
mod.crw_pf_rr <- route_path(mod.crw_pf,
                            what = "fitted",
                            map_scale = 10, 
                            dist = 10000
                            append = T,
                            buffer = 0.5,
                            centroids = TRUE,
                            )
```
setting `centroids = TRUE` essentially adds more resolution (& therefore more possible solutions) to the underlying mesh structure used to reroute the track. Setting a `buffer` distance > 0 moves the track a bit further away from any land barriers, which can (sometimes) result in a better rerouted solution. Some trial and error with the buffer distance (units in km) is usually needed.

Ian

**Reply by vheim** [reply]
Hi Ian,

Thank you for reply and sorry for the late answer - got caught up in another project. 

Your suggestions of adding `centroids = TRUE` and setting a `buffer` > 0 worked very well. I still lose some (very few data points), but when compared to the locations of the points I lose, it makes sense that the network of land barriers is too complex as you suggested. 

Best wishes,
Vital
